src/cqrs/read-event-bus.ts

```diff
@@ -0,0 +1,35 @@
+import { CommandBus, EventBus as Parent } from '@nestjs/cqrs';
+import { Injectable } from '@nestjs/common';
+import {
+  IMappedEventOptions,
+  IReadEvent,
+  IReadEventBusConfig,
+} from '../interfaces';
+import { defaultEventMapper } from './default-event-mapper';
+import { ModuleRef } from '@nestjs/core';
+
+@Injectable()
+export class ReadEventBus<
+  EventBase extends IReadEvent = IReadEvent
+> extends Parent<EventBase> {
+  constructor(
+    commandBus: CommandBus,
+    moduleRef: ModuleRef,
+    private readonly config: IReadEventBusConfig,
+  ) {
+    super(commandBus, moduleRef);
+  }
+  publish<T extends EventBase>(event: T) {
```

Comment:
Why do you need `publish` ans `publishAll` on the read event bus ?  
it's the input canal for saga and event handler ?

Reply:
We could use the standard event bus, but I want to add class validator to the events before publishing them. To be discussed